### PR TITLE
Add project errorlist and project info support in the tsserver.js APIs

### PR DESCRIPTION
**Pull request by zhengbli**
Those are APIs that sublime text plugin needs for features like build and project error list.

### CODE REVIEW

File: src/server/editorServices.ts [hunk 1]
```diff
@@ -309,6 +309,11 @@ module ts.server {
             return this.filenameToSourceFile[info.fileName];
         }
 
+        getFileNameList() {
```
Comment (CyrusNajmabadi) [inline src/server/editorServices.ts]: getFileNames?

File: src/server/protocol.d.ts [hunk 2]
```diff
@@ -817,6 +857,31 @@ declare module ts.server.protocol {
     }
     
     /**
+      * Arguments for geterrForProject request.
+      */
+    export interface GeterrForProjectRequestArgs {
```
Comment (CyrusNajmabadi) [inline src/server/protocol.d.ts]: "Geterr"?

Comment (zhengbli) [inline src/server/protocol.d.ts]: Just to be consistent with the other interface called ``GeterrRequestArgs``

File: src/server/session.ts [hunk 3]
```diff
@@ -677,6 +679,45 @@ module ts.server {
...
+            let lowPriorityFiles: string[] = [];
+            let veryLowPriorityFiles: string[] = [];
+            let normalizedFilePath = ts.normalizePath(fileName);
+            let project = this.projectService.getProjectForFile(normalizedFilePath);
+            for (let oneFile of fileNameList) {
+                if (oneFile == normalizedFilePath)
+                    highPriorityFiles.push(oneFile);
+                else {
+                    let info = this.projectService.getScriptInfo(oneFile);
+                    if (!info || !info.isOpen) {
```
Comment (mhegazy) [inline src/server/session.ts]: when would info be undefined?

File: src/server/session.ts [hunk 4]
```diff
@@ -677,6 +679,45 @@ module ts.server {
...
+                        if (oneFile.indexOf(".d.ts") > 0)
+                            veryLowPriorityFiles.push(oneFile);
+                        else
+                            lowPriorityFiles.push(oneFile);
+                        this.projectService.openFile(oneFile, true);
```
Comment (mhegazy) [inline src/server/session.ts]: i do not think we need this.. the editor will send open messages when the file is open.

File: src/server/session.ts [hunk 5]
```diff
@@ -677,6 +679,45 @@ module ts.server {
...
+
+            // Sort the file name list to make the recently touched files come first
+            let highPriorityFiles: string[] = [];
+            let mediumPriorityFiles: string[] = [];
+            let lowPriorityFiles: string[] = [];
+            let veryLowPriorityFiles: string[] = [];
+            let normalizedFilePath = ts.normalizePath(fileName);
+            let project = this.projectService.getProjectForFile(normalizedFilePath);
+            for (let oneFile of fileNameList) {
+                if (oneFile == normalizedFilePath)
```
Comment (mhegazy) [inline src/server/session.ts]: We should consider using FileMap to store these, and use its helpers to do the lookup. this comparison will fail if one of the files has the wrong slashes or casing,

Comment (zhengbli) [inline src/server/session.ts]: The ``FileMap`` only exists in master, and on the second thought that ``master`` would be a better place to merge for all project-diagnostics related code, as they are not needed for this sublime plugin release. Therefore closing this PR

Comment (zhengbli) [inline src/server/session.ts]: Thinking more into it I'm not sure how FileMap fits in here. As the only related operation is the one comparison. And the ``ts.normalizePath`` already includes normalize slashes. Let's discuss tomorrow.

**Pull request closed without merging**